inference-engine/tests/functional/plugin/conformance/subgraphs_dumper/include/matchers/base_matcher.hpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "pugixml.hpp"
+#include "ngraph/node.hpp"
+
+namespace SubgraphsDumper {
+
+class MatchersManager;
+
+class iMatcherConfig {
+public:
+    using Ptr = std::shared_ptr<iMatcherConfig>;
+
+    // Empty vectors stands for any of possible values
+    std::vector<std::string> ignored_attributes;
+    std::vector<size_t> ignored_ports;
+    bool is_fallback_config;
+
+    virtual bool op_in_config(const std::shared_ptr<ngraph::Node> &node) = 0;
+};
+
+template<typename ...OPTypes>
+struct MatcherConfig : public iMatcherConfig {
+public:
+    MatcherConfig() : iMatcherConfig() {
+        if (sizeof...(OPTypes) == 0) {
+            is_fallback_config = true;
+        }
+    }
+
+    MatcherConfig(const std::vector<std::string> &_ignored_attributes,
+                  const std::vector<size_t> &_ignored_ports) : iMatcherConfig() {
+        ignored_attributes = _ignored_attributes;
+        ignored_ports = _ignored_ports;
+
+        if (sizeof...(OPTypes) == 0) {
+            is_fallback_config = true;
+        }
+    }
+
+
+    bool op_in_config(const std::shared_ptr<ngraph::Node> &node) override {
+        std::initializer_list<bool> vals{(ngraph::is_type<OPTypes>(node))...};
+        return std::any_of(vals.begin(), vals.end(), [](bool i) { return i; });
+    };
+};
+
+class Matcher {
```

Comment:
IMHO, this class need `virtual` destructor.

Reply:
But what kind of resource management are you presume in this destructor? 
`Same question regarding destructor for iMatcherConfig`

Reply:
I expect something like this: `virtual ~Matcher() = default;`
This is more general rule if you have **interface** base class. It almost always should have virtual d-tor.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c127-a-class-with-a-virtual-function-should-have-a-virtual-or-protected-destructor

https://godbolt.org/z/75qnxr

Reply:
Virtual default destructor added